src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1381,6 +1381,20 @@
   (let [last-modified-milliseconds (.lastModified (io/file cacrl))]
        (time-coerce/from-long last-modified-milliseconds)))
 
+(schema/defn ^:always-validate update-crl :- CertificateRevocationList
+  [upstream-crl-pem :- InputStream
+   crl-path :- schema/Str]
+  (log/info (i18n/trs "Updating CRL at {0}" crl-path))
+  (let [upstream-crls (utils/pem->crls upstream-crl-pem)
+        our-crl (first (utils/pem->crls crl-path))
+        new-crl-chain (cons our-crl upstream-crls)]
+    ;; what verification do we need?
+    ;; do we need to confirm that a custom intermediate CA is being used?
+    ;; do we need to verify everything against the cert chain? i.e. confirm the order?
+    ;; do we need to check that our ca cert was signed by the subject of first crl in upstream-crls?
```

Comment:
iirc, the ordering of the certs and crls is simply a convention. One we rely on to load our cert, but I think in theory the rest of the CRLs could be in any order or contain other unrelated CRLs. So I'm not super into doing the order or quantity dependent checks you mention.

Reply:
I was hoping that whatever verification does would warn us if anything is actually awry, and call that good enough. But I haven't verified that that's actually true.

Reply:
Yes. Verifying an SSL context object like Justin suggests.

Reply:
kk, thanks